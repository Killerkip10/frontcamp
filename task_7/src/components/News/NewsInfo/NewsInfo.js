import React, { memo } from 'react';

export const NewsInfo = memo(({ info: { subsection, title, abstract, section } }) => (
  <div>
    <div>{subsection}</div>
    <div>{title}</div>
    <div>{abstract}</div>
    <div>{section}</div>
  </div>
));
