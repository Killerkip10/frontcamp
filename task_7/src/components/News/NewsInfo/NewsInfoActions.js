export const NEWS_INFO_GET_REQUEST = 'NEWS_INFO_GET_REQUEST';
export const NEWS_INFO_GET_SUCCESS = 'NEWS_INFO_GET_SUCCESS';
export const NEWS_INFO_GET_FAILURE = 'NEWS_INFO_GET_FAILURE';

export const CHANGE_TOPIC = 'CHANGE_TOPIC';

export const getNewsByIdRequest = () => ({
  type: NEWS_INFO_GET_REQUEST,
  payload: { isFetching: true },
});

export const getNewsByIdSuccess = info => ({
  type: NEWS_INFO_GET_SUCCESS,
  payload: { info, isFetching: false },
});

export const getNewsByIdFailure = () => ({
  type: NEWS_INFO_GET_FAILURE,
  payload: { isFetching: false },
})
