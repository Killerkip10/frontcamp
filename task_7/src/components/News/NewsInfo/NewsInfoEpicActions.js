import { API, API_KEY } from 'configs/api';

import {
  getNewsByIdRequest,
  getNewsByIdSuccess,
  getNewsByIdFailure,
} from './NewsInfoActions';

export const getNewsById = (index, topic) => (dispatch) => {
  dispatch(getNewsByIdRequest());

  return fetch(`${API.HOST}${API.GET_ARTICLES}${topic}.json?api-key=${API_KEY}`)
    .then(response => response.json())
    .then(({ results }) => dispatch(getNewsByIdSuccess(results[index])))
    .catch(() => dispatch(getNewsByIdFailure()));
};