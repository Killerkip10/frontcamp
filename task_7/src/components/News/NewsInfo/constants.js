export const DATA_CY = {
    LOADING: 'loading',
    ITEM_INFO: 'itemInfo',
};