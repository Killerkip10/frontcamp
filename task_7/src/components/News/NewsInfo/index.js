export * from './NewsInfoContainer';
export * from './NewsInfoReducer';  