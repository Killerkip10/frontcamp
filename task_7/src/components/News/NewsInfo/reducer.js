import {
  NEWS_INFO_GET_REQUEST,
  NEWS_INFO_GET_SUCCESS,
  NEWS_INFO_GET_FAILURE,
} from './actions';

const initialState = {
  isFetching: false,
  info: {},
};

export const newsInfo = (state = initialState, { type, payload }) => {
  switch (type) {
    case NEWS_INFO_GET_REQUEST:
    case NEWS_INFO_GET_SUCCESS:
    case NEWS_INFO_GET_FAILURE:
      return { ...state, ...payload };
    default:
      return state;
  }
};