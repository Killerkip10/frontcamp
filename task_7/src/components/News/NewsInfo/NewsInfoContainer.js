import React from 'react';
import { connect } from 'react-redux';

import { NewsInfo } from './NewsInfo';
import { getNewsById } from './epicActions';

export const NewsInfoContainerComponent = ({ isFetching, info, getNewsById }) => (
  <NewsInfo
    isFetching={isFetching}
    info={info}
    getNewsById={getNewsById}
  />
);

const mapStateToProps = state => ({
  isFetching: state.news.newsInfo.isFetching,
  info: state.news.newsInfo.info,
});

const mapStateToDispatch = (dispatch, { match: { params } }) => ({
  getNewsById: topic => dispatch(getNewsById(params.index, params.topic)),
});

const connector = connect(mapStateToProps, mapStateToDispatch);

export const NewsInfoContainer = connector(NewsInfoContainerComponent);