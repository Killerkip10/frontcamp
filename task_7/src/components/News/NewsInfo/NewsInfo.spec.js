import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { NewsInfo } from './NewsInfo';

describe('NewsInfo Component', () => {
    let props;

    beforeEach(() => {
        props = {
            info: {
                subsection: 'subsection-1',
                title: 'title-1',
                abstract: 'abstract-1',
                section: 'section-1',
            },
        };
    });

    it('', () => {
        const component = shallow(<NewsInfo {...props} />);

        expect(toJson(component)).toMatchSnapshot();
    });
});