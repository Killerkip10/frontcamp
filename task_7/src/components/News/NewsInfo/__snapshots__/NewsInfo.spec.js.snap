// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsInfo Component  1`] = `
<div
  data-cy="itemInfo"
>
  <div>
    subsection-1
  </div>
  <div>
    title-1
  </div>
  <div>
    abstract-1
  </div>
  <div>
    section-1
  </div>
</div>
`;
